localstack/services/sns/provider.py

```diff
@@ -169,9 +181,66 @@ def publish_message(
             message_attributes,
         )
     )
+
     return message_id
 
 
+def message_to_endpoint(
+    target_arn,
+    message,
+    structure,
+):
+    sns_client = aws_stack.connect_to_service("sns")
+    app_name = target_arn.split("/")[-2]
+    platform_name = target_arn.split("/")[-3]
+
+    try:
+        endpoint_attributes = sns_client.get_endpoint_attributes(EndpointArn=target_arn)[
+            "Attributes"
+        ]
+    except botocore.exceptions.ClientError:
+        LOG.warning(f"Missing attributes for endpoint: {target_arn}")
+        return
+
+    try:
+        platform_apps = sns_client.list_platform_applications()["PlatformApplications"]
+        app = [x for x in platform_apps if app_name in x["PlatformApplicationArn"]][0]
+    except IndexError:
+        LOG.warning(f"Missing Platform Application for endpoint: {target_arn}")
+        return
+
+    if structure == "json":
+        message = json.loads(message)
+
+    response = None
+    if platform_name == "GCM":
+        response = send_message_to_GCM(app["Attributes"], endpoint_attributes, message["GCM"])
+    # TODO: Add support for other platforms
+
+    if response is not None:
+        sns_backend = SNSBackend.get()
+        cache = sns_backend.platform_endpoint_responses[target_arn] = (
+            sns_backend.platform_endpoint_responses.get(target_arn) or []
+        )
+        cache.append({"status_code": response.status_code, "body": response.content})
```

Comment:
I think that when the POST request to GCM can’t go through, it should send a message to a SQS DLQ if there is one configured. It would allow us to remove the dicts with responses in the `SNSBackend`.

Reply:
now the failed publishing to a endpoint target triggers an error, so if the publishing comes from the sns trying to send a message to a subscriber it should be store in a SQS DLQ if there is one. (https://github.com/pinzon/localstack/blob/6d5671a7b38c3bc83b8ff0563d4a956b60b0523f/localstack/services/sns/provider.py#L940)

I deleted the dictionary since now I can make the test asserting the error.